src/app/MessageDef.h

```diff
@@ -0,0 +1,894 @@
+
+private:
+    EventPath::Builder mEventPathBuilder;
+};
+}; // namespace EventPathList
+
+namespace CommandPath {
+enum
+{
+    kCsTag_EndpointId          = 0,
+    kCsTag_GroupId             = 1,
+    kCsTag_NamespacedClusterId = 2,
+    kCsTag_CommandId           = 3,
+};
+
+class Parser : public ParserBase
+{
+public:
+    // aReader has to be on the element of EventPath container
+    CHIP_ERROR Init(const chip::TLV::TLVReader & aReader);
+
+    // Roughly verify the schema is right, including
+    // 1) all mandatory tags are present
+    // 2) no unknown tags
+    // 3) all elements have expected data type
+    // 4) any tag can only appear once
+    CHIP_ERROR CheckSchemaValidity() const;
+
+    // CHIP_END_OF_TLV if there is no such element
+    // CHIP_ERROR_WRONG_TLV_TYPE if there is such element but it's not any of the defined unsigned integer types
+    CHIP_ERROR GetEndpointId(chip::EndpointId * const apEndpointId) const;
+
+    // CHIP_END_OF_TLV if there is no such element
+    // CHIP_ERROR_WRONG_TLV_TYPE if there is such element but it's not any of the defined unsigned integer types
+    CHIP_ERROR GetGroupId(chip::GroupId * const apGroupId) const;
+
+    // CHIP_END_OF_TLV if there is no such element
+    // CHIP_ERROR_WRONG_TLV_TYPE if there is such element but it's not any of the defined unsigned integer types
+    CHIP_ERROR GetNamespacedClusterId(chip::ClusterId * const apNamespacedClusterId) const;
+
+    // CHIP_END_OF_TLV if there is no such element
+    // CHIP_ERROR_WRONG_TLV_TYPE if there is such element but it's not any of the defined unsigned integer types
+    CHIP_ERROR GetCommandId(chip::CommandId * const apCommandId) const;
+};
+
+class Builder : public BuilderBase
+{
+public:
+    CHIP_ERROR Init(chip::TLV::TLVWriter * const apWriter);
+    CHIP_ERROR Init(chip::TLV::TLVWriter * const apWriter, const uint8_t aContextTagToUse);
+
+    CommandPath::Builder & EndpointId(const chip::EndpointId aEndpointId);
+    CommandPath::Builder & GroupId(const chip::GroupId aGroupId);
+    CommandPath::Builder & NamespacedClusterId(const chip::ClusterId aNamespacedClusterId);
+    CommandPath::Builder & CommandId(const chip::CommandId aCommandId);
+
+    CommandPath::Builder & EndOfCommandPath();
+
+private:
+    CHIP_ERROR _Init(chip::TLV::TLVWriter * const apWriter, const uint64_t aTag);
+};
+}; // namespace CommandPath
+
+namespace EventDataElement {
+enum
+{
+    kCsTag_EventPath       = 0,
+    kCsTag_ImportanceLevel = 1,
+    kCsTag_Number          = 2,
+    kCsTag_UTCTimestamp    = 3,
+    kCsTag_SystemTimestamp = 4,
+    kCsTag_DeltaUTCTime    = 5,
+    kCsTag_DeltaSystemTime = 6,
+    kCsTag_Data            = 7,
+};
+
+class Parser : public ParserBase
+{
+public:
+    CHIP_ERROR Init(const chip::TLV::TLVReader & aReader);
+
+    // Roughly verify the schema is right
+    CHIP_ERROR CheckSchemaValidity() const;
+
+    CHIP_ERROR GetEventPath(EventPath::Parser * const apEventPath);
```

Comment:
were it me, I'd leave off the `Get` for these methods

Reply:
For example, if apply Get for these methods for same input type uint64_t, inside, their tag are different.
CHIP_ERROR EventDataElement::Parser::GetUTCTimestamp(uint64_t * const apUTCTimestamp)
{
    return GetUnsignedInteger(kCsTag_UTCTimestamp, apUTCTimestamp);
}

CHIP_ERROR EventDataElement::Parser::GetSystemTimestamp(uint64_t * const apSystemTimestamp)
{
    return GetUnsignedInteger(kCsTag_SystemTimestamp, apSystemTimestamp);
}